appinventor/appengine/src/com/google/appinventor/client/utils/Urls.java

```diff
@@ -48,11 +48,15 @@ private static boolean isSet(String str) {
    * repo, and galleryId parameters from the current session.
    *
    * @param base the base URI
+   * @param hasParam true if the base URL already has a parameter, so ? should not be used
    * @return the new URI to redirect to
    */
-  public static String makeUri(String base) {
+  public static String makeUri(String base, boolean hasParam) {
```

Comment:
This is okay as-is, but I'm wondering if we could just make this smarter since `hasParam=true iff base.contains("?")`?

Reply:
Deferred until later.